### ISSUE TITLE
Playbook-local vars_files loading a vault-encrypted file extremely slow

### ISSUE THREAD

**Issue by gplasky** [ansible/ansible#8340]
##### Issue Type:

Bug Report
##### Ansible Version:

```
$ ansible-playbook --version
ansible-playbook 1.7 (devel d51e10a3f4) last updated 2014/07/26 13:13:39 (GMT -700)
```
##### Environment:

Debian Squeeze
##### Summary:

Ref: https://github.com/ansible/ansible/pull/6734

Using the aforementioned code merged into devel, I tested `ansible-playbook` with a very simple play, operating on ~500 hosts in a single group. With a vault-encrypted file, when `group_vars` is used, playbook execution time is as expected however when used with `vars_files`, execution time skyrockets.
##### Steps To Reproduce:

Sample playbook used to test:

``` yml

---
# ansible-test.yml
- hosts: foo_group
  sudo: no
  gather_facts: no

  vars_files:
    - foo.yml

  tasks:
    - name: Noop
      command: uptime
```
##### Expected Results:

One would expect playbook execution to run in approximately the same amount of time when using encrypted vs unencrypted variable files.

Decrypted file with `vars_files`:

```
$ time ansible-playbook ansible-test.yml 2>&1 | grep real

real    0m23.729s
```
##### Actual Results:

Encrypted file with `vars_files`:

```
$ time ansible-playbook --ask-vault-pass 2>&1 | grep real
Vault password:

real    6m15.755s
```

_Note:_ With the deve version mentioned above, `group_vars` performs favorably whereas `vars_files` still does not. 

Remove `vars_files` and use `group_vars` within the playbook directory instead:

```
$ time ansible-playbook ansible-test.yml --ask-vault-pass 2>&1 | grep real
Vault password:

real    0m26.614s
```

**Comment by mpdehaan** [comment]
Ok this is probably loading vars_files once per instance, and as such is a request to load vars_files earlier if it doesn't depend on fact or inventory scoped variables in the vars_files path.

We can explore.

**Comment by davidbirdsong** [comment]
Any updates? A playbook of mine running over 100+ hosts takes 10-15 mins to make any progress once setup completes. It took a lot of eliminating of roles/tasks to narrow down to vault.

We plan to need ansible to run on way more than 100 hosts. Should I ditch vault?

**Comment by gplasky** [comment]
Based on my tests, using `group_vars` with vault instead of `vars_files` sufficed to keep the runtime within acceptable bounds. I imagine there would be cases where the use of the latter is necessary/desirable however if you can get away with the former, I'd suggest that route until this can be addressed.

**Comment by jctanner** [comment]
Testcase ...

```
[jtanner@fedmac AP-8340]$ cat inventory.py 
#!/usr/bin/env python

import json
import sys
from pprint import pprint

INV = {}
INV['_meta'] = {'hostvars': {}}

groups = ['group1']
hosts = []
for x in xrange(0, 100):
    hosts.append('host-%s' % x)

for idx, group in enumerate(groups):
    INV[group] = {}
    INV[group]['children'] = []
    INV[group]['vars'] = {}
    INV[group]['hosts'] = [x for x in hosts]

for host in hosts:
    INV['_meta']['hostvars'][host] = {}
    INV['_meta']['hostvars'][host]['ansible_connection'] = 'local'

print json.dumps(INV, indent=2)
```

```
[jtanner@fedmac AP-8340]$ cat site.yml 
- hosts: group1
  gather_facts: False
  vars_files:
    - foo.yml
  tasks:
    - debug: var=foo
```

```
[jtanner@fedmac AP-8340]$ cat foo.yml 
$ANSIBLE_VAULT;1.1;AES256
35636463656133313934393265366339303562626566666634303331626663633464376166323366
6234613932613138326233363234646533386563616161350a653436376332663565626439303066
32303262626365333635643837373134303934663863663738353630636636643762373166303166
3335303661323965340a336161353962383661383461356163356136333739313438636664346335
3836
[jtanner@fedmac AP-8340]$ cat vault.password 
redhat
```

```
[jtanner@fedmac AP-8340]$ cat test.sh 
#!/bin/bash
time ansible-playbook -i inventory.py --vault-password-file=vault.password site.yml
```

Results ...

```
1.7.2       1m16.857s
1.8.4       0m1.290s
1.9.6       0m2.018s        
2.0.0.2      0m8.973s
2.2.0       0m6.242s
```

**Comment by jctanner** [comment]
Based on the results above, this issue was fixed sometime during 1.8.x
